### PR TITLE
fixed i2c initialization

### PR DESCRIPTION
**Pull request by virvel**
Added missing initialization of I2CHandle. Should be working now. :)

### CODE REVIEW

File: src/dev/trill/I2c.h [hunk 1]
```diff
@@ -51,6 +51,7 @@ inline int I2c::initI2C_RW(int bus, int address, int dummy)
 	}
 	cfg.address = i2cAddress; // this seems unused anyhow
 	i2cAddress = address;
+	i2cHandle.Init(cfg);
```
Comment (dromer) [inline src/dev/trill/I2c.h:54]: yes! this is the main bit missing to get Trill to work!

Comment (dromer) [reply src/dev/trill/I2c.h:54]: @giuliomoro add this fix to open a PR for review to E-S?

**Comment by giuliomoro** [comment]
thanks and apologies for missing the earlier notifications. @virvel your fix of course looks great (silly me). I wonder what happens when you instantiate more than one device. Is it going to behave OK?

**Comment by virvel** [comment]
> thanks and apologies for missing the earlier notifications. @virvel your fix of course looks great (silly me). I wonder what happens when you instantiate more than one device. Is it going to behave OK?

great! no worries :) not sure about additional devices. could you make a pull request of this to electro-smith/libDaisy? maybe that was not your intention in the first place, but it would be nice to have it there. :)

**Comment by giuliomoro** [comment]
@virvel Sure, I'd be happy to make a PR there if you can confirm this works. I haven't had a chance to test it myself.

**Comment by virvel** [comment]
@giuliomoro It works using a single I2C device connected at least. I've been using the trill craft a lot with the daisy seed. :) I think @dromer  has been using it successfully as well.
There has been a few minor changes to some parts of libDaisy since you made this, but it's easily fixed.

**Comment by dromer** [comment]
@virvel not actively, mostly just did a couple tests since I have some Trill sensors that I'd like to use with libDaisy (hoping in combo with Heavy in the future).

This initialization fix at least allowed to using it as a PoC, but I think it would still need some work to properly integrate with libDaisy.

**Comment by giuliomoro** [comment]
What more work do you think is needed? The library itself is fully featured, but I understand that the example could have a bit more meat in it. Is there some libDaisy convention that I am missing ?

**Comment by dromer** [comment]
Shouldn't the device config and init not follow the libDaisy conventions for this class of peripheral device?

https://github.com/electro-smith/libDaisy/blob/master/src/dev/tlv493d.h#L35-L71
https://github.com/electro-smith/libDaisy/blob/master/src/dev/mpr121.h#L23-L60

**Comment by giuliomoro** [comment]
Right, that's all done here https://github.com/giuliomoro/libDaisy/blob/64c08470c079693cd9e2ab59cf2b85a5cff4727c/src/dev/trill/I2c.h

**Comment by giuliomoro** [comment]
I rationalised the commits, rebased and it's now a PR https://github.com/electro-smith/libDaisy/pull/657

**Comment by dromer** [comment]
I guess what I mean is that the interface with libDaisy should follow the common OOP approach, where the rest of the Trill implementation is more functional.

Does that make sense?

**Comment by giuliomoro** [comment]
turns out things have changed upstream so the rebase fails CI ... let's have further discussion there.

**Pull request merged**